presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/ShardMetadataRecordCursor.java

```diff
@@ -265,11 +274,22 @@ private ResultSet getNextResultSet()
 
         Long tableId = tableIds.next();
         Long columnId = metadataDao.getTemporalColumnId(tableId);
+        List<String> columnNames;
 
-        String minColumn = (columnId == null) ? "null" : minColumn(columnId);
-        String maxColumn = (columnId == null) ? "null" : maxColumn(columnId);
+        if (columnId == null) {
+            columnNames = getMappedColumnNames(DEFAULT_COLUMN_NAME, DEFAULT_COLUMN_NAME, DEFAULT_COLUMN_NAME, DEFAULT_COLUMN_NAME);
+        }
+        else {
+            Type temporalType = metadataDao.getTableColumn(tableId, columnId).getDataType();
+            checkState(temporalType.equals(DATE) || temporalType.equals(TIMESTAMP), "Temporal column should be of type date or timestamp: %s", temporalType);
+            if (temporalType.equals(DATE)) {
+                columnNames = getMappedColumnNames(DEFAULT_COLUMN_NAME, DEFAULT_COLUMN_NAME, minColumn(columnId), maxColumn(columnId));
+            }
+            else {
```

Comment:
I'd rather do it this way:
```
if(type.equals(DATE)){
..
} else if (type.equals(TIMESTAMP)) {
..
} else {
  throw exception
}
```
Then you can remove `checkState`. This way is less prone to errors when adding more clauses and builds in error check logic.